packages/contracts/src/dollar/amo/AaveAmo.sol

```diff
@@ -0,0 +1,272 @@
+        require(
+            _aaveRewardsController != address(0),
+            "Aave rewards controller address cannot be zero"
+        );
+        require(
+            _aavePoolDataProvider != address(0),
+            "Aave pool data provider address cannot be zero"
+        );
+
+        // Set contract owner
+        transferOwnership(_ownerAddress);
+
+        // Set the Amo minter
+        amoMinter = UbiquityAmoMinter(_amoMinterAddress);
+
+        // Set the Aave pool
+        aavePool = IPool(_aavePool);
+
+        // Set the Aave token
+        aaveToken = ERC20(_aaveToken);
+
+        // Set the Aave rewards controller
+        aaveRewardsController = IRewardsController(_aaveRewardsController);
+
+        // Set the Aave pool data provider
+        aavePoolDataProvider = IPoolDataProvider(_aavePoolDataProvider);
+    }
+
+    /* ========== Aave V3 + REWARDS ========== */
+
+    /**
+     * @notice Deposits collateral to Aave pool
+     * @param collateralAddress Address of the collateral ERC20
+     * @param amount Amount of collateral to deposit
+     */
+    function aaveDepositCollateral(
+        address collateralAddress,
+        uint256 amount
+    ) public onlyOwner {
+        ERC20 token = ERC20(collateralAddress);
+        token.safeApprove(address(aavePool), amount);
+        aavePool.deposit(collateralAddress, amount, address(this), 0);
+
+        emit CollateralDeposited(collateralAddress, amount);
+    }
+
+    /**
+     * @notice Withdraws collateral from Aave pool
+     * @param collateralAddress Address of the collateral ERC20
+     * @param aTokenAmount Amount of collateral to withdraw
+     */
+    function aaveWithdrawCollateral(
+        address collateralAddress,
+        uint256 aTokenAmount
+    ) public onlyOwner {
+        aavePool.withdraw(collateralAddress, aTokenAmount, address(this));
+
+        emit CollateralWithdrawn(collateralAddress, aTokenAmount);
+    }
+
+    /**
+     * @notice Borrows an asset from Aave pool
+     * @param asset Address of the asset to borrow
+     * @param borrowAmount Amount of asset to borrow
+     * @param interestRateMode Interest rate mode: 1 for Stable, 2 for Variable
+     */
+    function aaveBorrow(
+        address asset,
+        uint256 borrowAmount,
+        uint256 interestRateMode
+    ) public onlyOwner {
+        aavePool.borrow(
+            asset,
+            borrowAmount,
+            interestRateMode,
+            0,
+            address(this)
+        );
+        aaveIsAssetBorrowed[asset] = true;
+        aaveBorrowedAssets.push(asset);
+
+        emit Borrowed(asset, borrowAmount, interestRateMode);
+    }
+
+    /**
+     * @notice Repays a borrowed asset to Aave pool
+     * @param asset Address of the asset to repay
+     * @param repayAmount Amount of asset to repay
+     * @param interestRateMode Interest rate mode: 1 for Stable, 2 for Variable
+     */
+    function aaveRepay(
```

Comment:
I think both `aaveBorrow()` and `aaveRepay()` can be removed since we don't plan to borrow against collateral in the `UbiquityPool`.

Reply:
do you prefer to keep 2 versions of it, one with borrow and another without, to have both just in case?

Reply:
What do you mean by "2 versions of it"?

Reply:
Keep fully capable (with borrow and repay) AMO contract as `AaveBorrowingAmo` under deprecated folder, and the one without borrowing as `AaveAmo` or `AaveSupplyingAmo`.  That is if borrowing against collateral, or looping/leveraging comes in handy, the contract will already be written.

Reply:
I've refactored to remove borrowing. If needed later can rollback to commit [ee11e95](https://github.com/ubiquity/ubiquity-dollar/pull/956/commits/ee11e957aab78f39fcb37192a8362527c199eb5a)